relay-server/src/actors/global_config.rs

```diff
@@ -0,0 +1,153 @@
+    fetch_interval: u64,
+}
+
+/// A way to get updates of the global config.
+pub enum GlobalConfigInterface {
+    /// Returns the most recent global config.
+    Get(Sender<Arc<GlobalConfig>>),
+    /// Returns a [`watch::Receiver`] where global config updates will be sent to.
+    Subscribe(Sender<watch::Receiver<Arc<GlobalConfig>>>),
+}
+
+impl Interface for GlobalConfigInterface {}
+
+/// The message for requesting the most recent global config from [`GlobalConfigService`].
+pub struct Get;
+
+impl FromMessage<Get> for GlobalConfigInterface {
+    type Response = AsyncResponse<Arc<GlobalConfig>>;
+
+    fn from_message(_: Get, sender: Sender<Arc<GlobalConfig>>) -> Self {
+        Self::Get(sender)
+    }
+}
+
+/// The message for receiving a watch that subscribes to the [`GlobalConfigService`].
+pub struct Subscribe;
+
+impl FromMessage<Subscribe> for GlobalConfigInterface {
+    type Response = AsyncResponse<watch::Receiver<Arc<GlobalConfig>>>;
+
+    fn from_message(_: Subscribe, sender: Sender<watch::Receiver<Arc<GlobalConfig>>>) -> Self {
+        Self::Subscribe(sender)
+    }
+}
+
+impl GlobalConfigService {
+    /// Creates a new [`GlobalConfigService`].
+    pub fn new(upstream: Addr<UpstreamRelay>) -> Self {
+        let (sender, _) = watch::channel(Arc::new(GlobalConfig::default()));
+        Self {
+            sender: Arc::new(sender),
+            upstream,
+            fetch_interval: 10,
+        }
+    }
+
+    fn handle_message(&self, message: GlobalConfigInterface) {
+        match message {
+            GlobalConfigInterface::Get(sender) => {
+                sender.send(self.sender.borrow().clone());
+            }
+            GlobalConfigInterface::Subscribe(sender) => {
+                sender.send(self.sender.subscribe());
+            }
+        }
+    }
+
+    /// Requests a new global config from upstream.
+    ///
+    /// This function does not check any authentication since the upstream
+    /// service deals with it (including observability):
+    /// - If Relay is authenticated, requests will complete as expected.
+    /// - If Relay isn't authenticated yet, the following request to successfull
+    /// authentication will happen.
+    /// - If Relay is permanently blocked, the upstream service will error on
+    /// requests.
+    fn update_global_config(&self) {
+        let upstream_relay: Addr<UpstreamRelay> = self.upstream.clone();
+        let sender = Arc::clone(&self.sender);
+
+        tokio::spawn(async move {
+            let query = GetProjectStates {
+                public_keys: vec![],
+                full_config: false,
+                no_cache: false,
+                global: true,
+            };
+
+            match upstream_relay.send(SendQuery(query)).await {
+                Ok(res) => Self::handle_upstream_response(res, sender),
+                Err(_) => relay_log::error!(
+                    "GlobalConfigService failed to send request to UpstreamService"
```

Comment:
_nit_: You can shorten the error message here as the logger will automatically add the module and location.